vcf.c

```diff
@@ -3204,12 +3439,24 @@ int vcf_parse(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v)
     khint_t k;
     ks_tokaux_t aux;
 
+//#define NOT_DOT(p) strcmp((p), ".")
+//#define NOT_DOT(p) (!(*p == '.' && !p[1]))
+//#define NOT_DOT(p) ((*p) != '.' || (p)[1])
+//#define NOT_DOT(p) (q-p != 1 || memcmp(p, ".\0", 2))
+#define NOT_DOT(p) (memcmp(p, ".\0", 2))
+
     if (!s || !h || !v || !(s->s))
         return ret;
 
     // Assumed in lots of places, but we may as well spot this early
     assert(sizeof(float) == sizeof(int32_t));
 
+    // Ensure string we parse has space to permit some over-flow when during
+    // parsing.  Eg to do memcmp(key, "END", 4) in vcf_parse_info over
+    // the more straight forward looking strcmp, giving a speed advantage.
+    if (ks_resize(s, s->l+4) < 0)
```

Comment:
Strictly this doesn't completely avoid UB because the guard bytes are uninitialised.  Although you may be saved by `memcmp()`'s behaviour of stopping on the first mismatching byte.